project3/coordinator/coordinator.cpp

```diff
@@ -18,30 +19,46 @@ Coordinator::Coordinator(
     std::shared_ptr<Registrar> registrar,
     std::shared_ptr<queue::Queue<MessageLog::Message>> msg_queue,
     std::shared_ptr<MessageLog> msg_log)
-    : msg_queue_(msg_queue),
-      msg_log_(msg_log),
+    : msg_queue_(std::move(msg_queue)),
+      msg_log_(std::move(msg_log)),
       client_fd_(participant_fd),
-      hostname_(hostname),
-      msg_mgr_(msg_mgr),
-      registrar_(registrar){}
+      hostname_(std::move(hostname)),
+      messenger_mgr_(std::move(msg_mgr)),
+      registrar_(std::move(registrar)) {}
 
 bool Coordinator::Coordinate() {
   ClientState client_state = ClientState::ACTIVE;
 
-  while (client_state == ClientState::ACTIVE) {
-    // Get the next message from the socket.
-    CoordinatorMessage message;
-    client_state = ReadNextMessage(&message);
+  // Get the message from the socket.
+  CoordinatorMessage message;
+  client_state = ReadNextMessage(&message);
 
-    if (client_state == ClientState::ACTIVE) {
-      // Handle the message.
-      client_state = DispatchMessage(message);
-    }
+  if (client_state == ClientState::ACTIVE) {
+    // Handle the message.
+    client_state = DispatchMessage(message);
   }
 
   return client_state != ClientState::ERROR;
 }
 
+void Coordinator::DetermineParticipant() {
+  // determine the participant.
+  for (auto participant :
+      registrar_->GetConnectedParticipants()->GetParticipants()) {
+    if (participant.hostname == hostname_) {
+      participant_ = participant;
+    }
+  }
```

Comment:
It would make me feel better to have a `CHECK` here making sure you didn't go through the loop without modifying anything.

Reply:
I think this should be resolved since I had these functions return their respective types to remove the `message_` and `participant_` members.